optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -58,55 +45,283 @@ that uses a functional approach:
 However, that scales poorly because it doesn't do an <<incrementalScoreCalculation,incremental calculation>>:
 When the planning variable of a single `Talk` changes, to recalculate the score,
 the normal Streams API has to execute the entire stream from scratch.
+The ConstraintStreams API allows you to write similar code in pure Java, while reaping the performance benefits of
+incremental score calculation.
+This is an example of the same code, using the Constraint Streams API:
 
-On the other hand, the ConstraintStreams API uses <<incrementalScoreCalculation,deltas>>:
+// TODO Rewrite the above paragraph PLANNER-1709 is fixed, mentioning constraint justification as another benefit.
 
-////
-TODO Replace the previous sentence when PLANNER-1709 is fixed.
+[source,java,options="nowrap"]
+----
+    private Constraint doNotAssignAnn(ConstraintFactory factory) {
+        return factory.from(Shift.class)
+                .filter(Shift::isEmployeeAnn)
+                .penalize("Don't assign Ann", HardSoftScore.ONE_SOFT);
+    }
+----
 
-So it doesn't do <<incrementalScoreCalculation,deltas>>.
-Nor does it support <<explainingTheScore,justifications>>.
+This constraint stream goes over all instances of class `Shift` amongst <<problemFacts,problem facts>> and
```

Comment:
grammar: "the" missing before "problem facts"

"amongst" -> maybe "among all problem facts"